fableous-fe/src/containers/ClassroomListPage.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  CircularProgress,
+  Grid,
+  Icon,
+  IconButton,
+  Typography,
+} from "@material-ui/core";
+import { Alert } from "@material-ui/lab";
+import useAxios from "axios-hooks";
+import { Formik } from "formik";
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import * as yup from "yup";
+import { restAPI } from "../Api";
+import FormikTextField from "../components/FormikTextField";
+import { APIResponse, Classroom } from "../Data";
+
+export default function ClassroomListPage() {
+  const [creating, setCreating] = useState(false);
+  const [
+    { data: classrooms, loading: getLoading, error: getError },
+    executeGet,
+  ] = useAxios<APIResponse<Classroom[]>, APIResponse<undefined>>(
+    restAPI.classroom.getList(),
+    { manual: true }
+  );
+  const [{ loading: postLoading }, executePost] = useAxios<
+    APIResponse<string>,
+    APIResponse<undefined>
+  >(restAPI.classroom.create(), {
+    manual: true,
+  });
+
+  const handleCreateSubmit = (classroom: Classroom) => {
+    executePost({
+      data: {
+        name: classroom.name.trim(),
+      },
+    })
+      .then((resp) => {
+        // eslint-disable-next-line no-param-reassign
+        classroom.id = resp.data.data || "";
+        classrooms?.data?.push(classroom);
+        setCreating(false);
+      })
+      .catch((error) => console.error(error));
+  };
+
+  const handleCreate = () => {
+    setCreating(true);
+  };
+
+  const handleCancel = () => {
+    setCreating(false);
+  };
+
+  useEffect(() => {
```

Comment:
whats the purpose of this useEffect?

Reply:
enforces classroom list to be refetched upon component reload

Reply:
so that when going back to list page after editing classroom name, classroom list is refeched